pilot/pkg/kube/inject/webhook.go

```diff
@@ -435,6 +444,10 @@ func createPatch(pod *corev1.Pod, prevStatus *SidecarInjectionStatus, annotation
 	patch = append(patch, addVolume(pod.Spec.Volumes, sic.Volumes, "/spec/volumes")...)
 	patch = append(patch, addImagePullSecrets(pod.Spec.ImagePullSecrets, sic.ImagePullSecrets, "/spec/imagePullSecrets")...)
 
+	if sic.DNSConfig != nil {
+		patch = append(patch, addPodDNSConfig(sic.DNSConfig, "/spec/dnsConfig")...)
```

Comment:
Will it add the search terms to the dnsConfig with this append?
Reason I'm asking is because I think the DNS suffixes should be under `/spec/dnsConfig/searches`. E.g.:
```yaml
dnsConfig:
    searches:
      - global
      - foo.global
```

Reply:
I did print out the pod data structure during testing.  Perhaps something has changed in the manifest after I rebased.  Will get back to you with more information.

Reply:
The answer is yes:
https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/templates/sidecar-injector-configmap.yaml#L288-L291

all dnsConfig values can be overridden in the sidecar injector now.  The manifest only overrides dnsConfig.searches.

Reply:
If you tested it and injected pod looks fine then it's lgtm.
Btw, what if you already have `dnsConfig` in the original Pod? Will the global be nicely appended?

Reply:
I hadn't considered the case of a pod having dnsConfig already present.  No clue what happens in that scenario.  If you think this is likely - can test it and address any work needed in a followup pr.  I don't expect many people to override their dncConfig values in their app manifests - although it could happen.

Cheers
-steve

Reply:
FYI, the existing code does not override the DNS config. according to kube docs, the DNS searches if specified will be appended to the default set of dns search namespaces for a pod.

Reply:
I did test this scenario this morning and the sidecar injector dnsConfig isn't applied if any dnsConfig in the pod.  See:
```
diff --git a/samples/sleep/sleep.yaml b/samples/sleep/sleep.yaml
index 36a90ca..f4874a2 100644
--- a/samples/sleep/sleep.yaml
+++ b/samples/sleep/sleep.yaml
@@ -50,4 +50,7 @@ spec:
         image: pstauffer/curl
         command: ["/bin/sleep", "3650d"]
         imagePullPolicy: IfNotPresent
+      dnsConfig:
+        searches:
+        - sdake.io
 ---
```

```
sdake@beast-01:~/go/src/istio.io/istio$ kubectl exec -it sleep-7c547f7f58-s42z2 -c sleep /bin/sh
/ # cat /etc/resolv.conf
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local sdake.io
options ndots:5
```

Will follow this up with a PR that handles this scenario.

Cheers
-steve